server/src/test/java/org/opensearch/ingest/AbstractBatchingProcessorTests.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.ingest;
+
+import org.opensearch.test.OpenSearchTestCase;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Consumer;
+
+public class AbstractBatchingProcessorTests extends OpenSearchTestCase {
+
+    public void testBatchExecute_emptyInput() {
+        DummyProcessor processor = new DummyProcessor(3);
+        Consumer<List<IngestDocumentWrapper>> handler = (results) -> assertTrue(results.isEmpty());
+        processor.batchExecute(Collections.emptyList(), handler);
+        assertTrue(processor.getSubBatches().isEmpty());
+    }
+
+    public void testBatchExecute_singleBatchSize() {
+        DummyProcessor processor = new DummyProcessor(3);
+        List<IngestDocumentWrapper> wrapperList = Arrays.asList(
+            IngestDocumentPreparer.createIngestDocumentWrapper(1),
+            IngestDocumentPreparer.createIngestDocumentWrapper(2),
+            IngestDocumentPreparer.createIngestDocumentWrapper(3)
+        );
+        List<IngestDocumentWrapper> resultList = new ArrayList<>();
+        processor.batchExecute(wrapperList, resultList::addAll);
+        assertEquals(wrapperList, resultList);
+        assertEquals(1, processor.getSubBatches().size());
+        assertEquals(wrapperList, processor.getSubBatches().get(0));
+    }
+
+    public void testBatchExecute_multipleBatches() {
+        DummyProcessor processor = new DummyProcessor(2);
+        List<IngestDocumentWrapper> wrapperList = Arrays.asList(
+            IngestDocumentPreparer.createIngestDocumentWrapper(1),
+            IngestDocumentPreparer.createIngestDocumentWrapper(2),
+            IngestDocumentPreparer.createIngestDocumentWrapper(3),
+            IngestDocumentPreparer.createIngestDocumentWrapper(4),
+            IngestDocumentPreparer.createIngestDocumentWrapper(5)
+        );
+        List<IngestDocumentWrapper> resultList = new ArrayList<>();
+        processor.batchExecute(wrapperList, resultList::addAll);
+        assertEquals(wrapperList, resultList);
+        assertEquals(3, processor.getSubBatches().size());
+        assertEquals(wrapperList.subList(0, 2), processor.getSubBatches().get(0));
+        assertEquals(wrapperList.subList(2, 4), processor.getSubBatches().get(1));
+        assertEquals(wrapperList.subList(4, 5), processor.getSubBatches().get(2));
+    }
+
+    public void testBatchExecute_randomBatches() {
```

Comment:
Could we use camel-case to name the method?

Reply:
Test cases use function names to express test scenarios which can be quite long, to have better readability, I prefer this naming style with "test{FunctionName}_{test scenario+expectation}". If we migrate to JUnit5, we can use `@DisplayName` annotation on each test case to have a clear explanation on the test case, then naming will not be that important.

https://www.baeldung.com/java-unit-testing-best-practices

Reply:
I think the prevailing style in the project is to not use underscores, but I don't have a problem with it and I'm there are probably existing examples that match this style.